### PR TITLE
libvirt: Fix shell reload

### PR DESCRIPTION
**Pull request by stevenhorsman**
I experimented with moving the shell reload straight after the pipx ensurepath when discussing this with Magnus, but in the GHA once merged this doesn't seem to be working as the reload means the script ends, so we need to move this back.

### CODE REVIEW

**Comment by mkulke** [comment]
I thought we concluded that launching a shell at the end of a script will make some sense on a local developer box, but if invoked in GHA, in a step without a tty, the shell would just immediately exit? 

can you point to the failed run maybe?

**Comment by stevenhorsman** [comment]
The latest daily e2e test is the failed run: https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/12356330516 and it's passing in this one. Sorry, I thought I'd reverted this after our discussion about it. I should have re-checked before I merged.

**Review by mkulke — APPROVED** [review]

File: src/cloud-api-adaptor/libvirt/config_libvirt.sh
Comment (wainersm) [inline src/cloud-api-adaptor/libvirt/config_libvirt.sh:89]: Running `exec $SHEL` here on gha will definitively stop here and do not run important setup steps (like creating the ssh key-pair).

Sorry, I missed the first discussion about that exec... would make sense to append `$HOME/.local/bin` to system's PATH if in gha action as shown in `https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#adding-a-system-path`

Comment (stevenhorsman) [reply src/cloud-api-adaptor/libvirt/config_libvirt.sh:89]: I did that in the previous PR: https://github.com/confidential-containers/cloud-api-adaptor/blob/8f534e3e27f81d4794d8bd548a276a67fad99773/.github/workflows/e2e_libvirt.yaml#L125

Comment (mkulke) [reply src/cloud-api-adaptor/libvirt/config_libvirt.sh:89]: but yes, @wainersm is right. if we pivot to a new shell mid-script, the rest of the script will be discarded in any case (locally or on the CI). There are a gazillion ways to update the path, but we should probably focus on unblocking the CI.

one of those options would be to have pipx installed in an outside step, the next steps starts a new shell and should have the path set. 

```yaml
name: install pipx
if: ... # onyl on ubuntu
run: |     
  sudo apt-get update
  sudo DEBIAN_FRONTEND=noninteractive apt-get  install pipx -y
  pipx ensurepath
name: run config-libvirt
run: ...  
```

Comment (stevenhorsman) [reply src/cloud-api-adaptor/libvirt/config_libvirt.sh:89]: I definitely understand the approach is probably nicer, but I would prefer to not move pipx in a separate step as I'd like to move more towards moving things from the workflow into shell scripts for easier PR testing than the opposite.

I've tried adding a new commit with `"${CI}" != "true"` which should mean we don't run the reload at all on GHA. Hopefully that is sufficient to get this PR merged and unblock the CI and I'm happy to discuss further improvements once we are back green again.

Comment (mkulke) [reply src/cloud-api-adaptor/libvirt/config_libvirt.sh:89]: I think the original approach of opening a new shell at the end of the script should at least do no harm. me, as an interactive user of a script, won't like this, but that's maybe idiosyncrasy

File: src/cloud-api-adaptor/libvirt/config_libvirt.sh
Comment (wainersm) [inline src/cloud-api-adaptor/libvirt/config_libvirt.sh:142]: Hi @stevenhorsman ,

One last fix, use `${CI:-}` to make `set -o nounset` happy.

Comment (stevenhorsman) [reply src/cloud-api-adaptor/libvirt/config_libvirt.sh]: It is interesting that the GHA shellcheck didn't pick up on that, maybe that's something we need to be aware of

Comment (wainersm) [reply src/cloud-api-adaptor/libvirt/config_libvirt.sh]: hmmm... are we shell-checking the scripts on workflows or all scripts?

**Pull request merged**
